src/main/java/de/undercouch/actson/JsonParser.java

```diff
@@ -399,14 +402,40 @@ private void parse(char nextChar) {
         event1 = stateToEvent();
       }
 
-      // Change the state.
```

Comment:
It states obvious. There would be some merit in it if this was unusual way or place to change the state. Neither is the case here. Also I do the same just below it twice more so it would cause additional confusion if some assignments had this comment while others - don't.